### PR TITLE
Make keras policy use max_history parameter in train.py

### PR DESCRIPTION
**Pull request by Ghostvv**
**Proposed changes**:
- fix: keras policy was not picking max_history parameter

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

**Comment by Ghostvv** [comment]
all the tests are passed. The main problem was that `train_dialogue_model(...)` was called wrongly. There was one parameter missing. The change with max_history is just for convenience, I think it is more logical for a featurizer to store its default value and not a policy

**Comment by Ghostvv** [comment]
I'll add explicit names for arguments in tests

File: rasa_core/agent.py [hunk 1]
```diff
@@ -197,8 +197,7 @@ def load_data(self,
         max_max_history = 0
         all_max_history_featurizers = True
         for policy in self.policy_ensemble.policies:
-            if (hasattr(policy, 'featurizer') and
-                    hasattr(policy.featurizer, 'max_history')):
+            if hasattr(policy.featurizer, 'max_history'):
```
[the remark this answers is not in the document]
Comment (Ghostvv) [reply rasa_core/agent.py]: so we explicitly set `FallbackPolicy.featurizer` to None, but it doesn't have attribute `max_history`. I checked locally, it worked

[the remark this answers is not in the document]
Comment (rusiaaman) [reply rasa_core/agent.py]: Yes. Sorry, I misread the change. Thanks.

**Pull request merged**